OutOfSchool/OutOfSchool.WebApi/Services/ProviderService.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.WebApi.Mapping.Extensions;
+using OutOfSchool.WebApi.Models;
+
+namespace OutOfSchool.WebApi.Services
+{
+    /// <summary>
+    /// Implements the interface with CRUD functionality for Provider entity.
+    /// </summary>
+    public class ProviderService : IProviderService
+    {
+        private readonly IProviderRepository repository;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProviderService"/> class.
+        /// </summary>
+        /// <param name="repository">Repository for some entity.</param>
+        public ProviderService(IProviderRepository repository)
+        {
+            this.repository = repository;
+        }
+
+        /// <inheritdoc/>
+        public Task<ProviderDto> Create(ProviderDto dto)
+        {
+            if (dto == null)
+            {
+                throw new ArgumentNullException(nameof(dto), "Provider was null.");
+            }
+
+            var provider = dto.ToDomain();
+
+            if (repository.Exists(provider))
+            {
+                throw new ArgumentException("There is already an providerDto with such data");
```

Comment:
I believe we should not write 'providerDto' here
it is more clear for the user to see just 'provider'